pkg/release/externals.go

```diff
@@ -0,0 +1,239 @@
+		}
+		return stringPtr("{UNITY_PATH}")
+	case "standardassets":
+		if platform == "darwin" {
+			return stringPtr("{UNITY_PATH}/Standard Assets")
+		}
+
+		return stringPtr("{UNITY_PATH}/Editor")
+	case "exampleprojects":
+		fallthrough
+	case "example":
+		return nil
+	case "android":
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer")
+	case "android-sdk-build-tools":
+		if ext == "zip" {
+			return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/SDK/build-tools")
+		}
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/SDK")
+	case "android-sdk-platforms":
+		if ext == "zip" {
+			return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/SDK/platforms")
+		}
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/SDK")
+	case "android-sdk-platform-tools":
+		fallthrough
+	case "android-sdk-ndk-tools":
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/SDK")
+	case "android-ndk":
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/NDK")
+	case "android-open-jdk":
+		return editorDataPath(platform, "PlaybackEngines/AndroidPlayer/OpenJDK")
+	case "ios":
+		return stringPtr("{UNITY_PATH}/PlaybackEngines")
+	case "tvos":
+		fallthrough
+	case "appletv":
+		return editorDataPath(platform, "PlaybackEngines/AppleTVSupport")
+	case "linux":
+		fallthrough
+	case "linux-mono":
+		fallthrough
+	case "linux-il2cpp":
+		return editorDataPath(platform, "PlaybackEngines/LinuxStandaloneSupport")
+	case "mac":
+		fallthrough
+	case "mac-mono":
+		fallthrough
+	case "mac-il2cpp":
+		return stringPtr("{UNITY_PATH}/Unity.app/Contents/PlaybackEngines/MacStandaloneSupport")
+	case "samsungtv":
+		fallthrough
+	case "samsung-tv":
+		return editorDataPath(platform, "PlaybackEngines/STVPlayer")
+	case "tizen":
+		return editorDataPath(platform, "PlaybackEngines/TizenPlayer")
+	case "vuforia":
+		fallthrough
+	case "vuforia-ar":
+		return editorDataPath(platform, "PlaybackEngines/VuforiaSupport")
+	case "webgl":
+		return editorDataPath(platform, "PlaybackEngines/WebGLSupport")
+	case "windows":
+		fallthrough
+	case "windows-mono":
+		fallthrough
+	case "windows-il2cpp":
+		return editorDataPath(platform, "PlaybackEngines/WindowsStandaloneSupport")
+	case "facebook":
+		fallthrough
+	case "facebook-games":
+		return editorDataPath(platform, "PlaybackEngines/Facebook")
+	case "facebookgameroom":
+		return nil
+	case "lumin":
+		return editorDataPath(platform, "PlaybackEngines/LuminSupport")
+	}
+
+	if strings.HasPrefix(name, "language-") {
+		if platform == "darwin" {
+			return stringPtr("{UNITY_PATH}/Unity.app/Contents/Localization")
+		}
+		return editorDataPath(platform, "Localization")
+	}
+
+	return stringPtr("{UNITY_PATH}")
+}
+
+func prepareExternalModule(platform string, m *ModuleRelease) {
+	ext := path.Ext(m.DownloadURL)
+	m.InstallOptions.Destination = moduleDestination(platform, m.ID, ext)
+}
+
+func generateAndroidModules(release *EditorRelease, platform string) {
```

Comment:
this function _looks_ unpleasant to maintain, in terms of 3rd party breaking assumptions over time - is this more stable than I fear?

Reply:
No, I wish there was anything else I could do here.